prime-angular-frontend/src/app/modules/site-registration/pages/organization-claim-page/organization-claim-page.component.ts

```diff
@@ -0,0 +1,116 @@
+import { Party } from '@lib/models/party.model';
+
+import { AuthService } from '@auth/shared/services/auth.service';
+import { BcscUser } from '@auth/shared/models/bcsc-user.model';
+import { SiteRoutes } from '@registration/site-registration.routes';
+import { OrganizationFormStateService } from '@registration/shared/services/organization-form-state.service';
+import { Organization } from '@registration/shared/models/organization.model';
+import { OrganizationClaimPageFormState } from './organization-claim-page-form-state.class';
+import { OrgnizationType } from '@registration/shared/models/organization-type-model';
+
+@Component({
+  selector: 'app-organization-claim-page',
+  templateUrl: './organization-claim-page.component.html',
+  styleUrls: ['./organization-claim-page.component.scss']
+})
+export class OrganizationClaimPageComponent extends AbstractEnrolmentPage implements OnInit {
+  public formState: OrganizationClaimPageFormState;
+  public title: string;
+  public routeUtils: RouteUtils;
+  public isCompleted: boolean;
+  public isClaimExistingOrg: boolean;
+  public hasOrgClaimError: boolean;
+  public busy: Subscription;
+
+  constructor(
+    protected dialog: MatDialog,
+    protected formUtilsService: FormUtilsService,
+    private organizationFormStateService: OrganizationFormStateService,
+    private organizationResource: OrganizationResource,
+    private toastService: ToastService,
+    private authService: AuthService,
+    private route: ActivatedRoute,
+    router: Router
+  ) {
+    super(dialog, formUtilsService);
+
+    this.title = this.route.snapshot.data.title;
+    this.routeUtils = new RouteUtils(route, router, SiteRoutes.MODULE_PATH);
+  }
+
+  public onClaimOrgChange(event: MatSlideToggleChange): void {
+    this.isClaimExistingOrg = event.checked;
+    this.toggleClaimFormValidators(event.checked);
+  }
+
+  public ngOnInit(): void {
+    this.createFormInstance();
+    this.isClaimExistingOrg = !!this.formState.json.pec && !!this.formState.json.claimDetail;
+    this.toggleClaimFormValidators(this.isClaimExistingOrg);
+  }
+
+  protected createFormInstance(): void {
+    this.formState = this.organizationFormStateService.organizationClaimPageFormState;
+  }
+
+  protected patchForm(): void {
+  }
+
+  protected performSubmission(): Observable<number> {
+    return this.authService.getUser$()
+      .pipe(
+        exhaustMap((bcscUser: BcscUser) => this.organizationResource.getSigningAuthorityByUserId(bcscUser.userId)),
+        exhaustMap((party: Party) => {
+          if (this.isClaimExistingOrg) {
+            return this.organizationResource.claimOrganization(party.id, this.formState.json)
+              .pipe(
+                catchError((error, caught) => {
+                  this.hasOrgClaimError = true;
+                  return of(0);
+                })
+            );
+          }
+          return this.organizationResource.createOrganization(party.id)
```

Comment:
We want to create the organization just in time.  You'll want to put organization creation on the OrganizationNamePage and have it create if the `oid` param doesn't exist (zero aka falsey), or update if the `oid` param exists (not zero aka truthy).